Content.Server/TapeRecorder/TapeRecorderComponent.cs

```diff
@@ -0,0 +1,64 @@
+using Content.Shared.TapeRecorder;
+using Robust.Shared.Audio;
+
+namespace Content.Server.TapeRecorder
+{
+    /// <summary>
+    /// Component given to tape recorders, will allow you to record to the stored cassette tape.
+    /// </summary>
+    [RegisterComponent]
+    public sealed class TapeRecorderComponent : Component
+    {
+
+        public CassetteTapeComponent? InsertedTape = null;
+
+        //Starts in record mode since it'll probably be an empty tape
+        public TapeRecorderState CurrentMode { get; set; } = TapeRecorderState.Empty;
+
+        public bool Enabled = false;
+
+        /// <summary>
+        ///AccumulatedTime recording was started on
+        /// </summary>
+        public float RecordingStartTime;
+
+        /// <summary>
+        ///The timestamp of the tape that we started recording on
+        /// </summary>
+        public float RecordingStartTimestamp;
```

Comment:
There is definitely better ways of getting time stamps and stuff. You probably want to ask Sloth.